lib/FSi/DoctrineExtensions/Translatable/ObjectAssociationContext.php

```diff
@@ -0,0 +1,147 @@
+     */
+    private $object;
+
+    /**
+     * @var PropertyAccessor
+     */
+    private $propertyAccessor;
+
+    /**
+     * @param ObjectManager $objectManager
+     * @param TranslationAssociationMetadata $associationMetadata
+     * @param object $object
+     */
+    public function __construct(
+        ObjectManager $objectManager,
+        TranslationAssociationMetadata $associationMetadata,
+        $object
+    ) {
+        $this->objectManager = $objectManager;
+        $this->associationMetadata = $associationMetadata;
+        $this->object = $object;
+    }
+
+    /**
+     * @return ObjectManager
+     */
+    public function getObjectManager()
+    {
+        return $this->objectManager;
+    }
+
+    /**
+     * @return TranslationAssociationMetadata
+     */
+    public function getAssociationMetadata()
+    {
+        return $this->associationMetadata;
+    }
+
+    /**
+     * @return object
+     */
+    public function getObject()
+    {
+        return $this->object;
+    }
+
+    /**
+     * @return ClassMetadata
+     */
+    public function getClassMetadata()
+    {
+        return $this->objectManager->getClassMetadata(get_class($this->object));
+    }
+
+    /**
+     * @return Mapping\ClassMetadata
+     */
+    public function getTranslatableMetadata()
+    {
+        return $this->associationMetadata->getClassMetadata();
+    }
+
+    /**
+     * @return ClassMetadata
+     */
+    public function getTranslationClassMetadata()
+    {
+        $meta = $this->getClassMetadata();
+        $assocName = $this->associationMetadata->getAssociationName();
+
+        return $this->objectManager->getClassMetadata($meta->getAssociationTargetClass($assocName));
+    }
+
+    /**
+     * @return string
+     */
+    public function getObjectLocale()
+    {
+        $localeProperty = $this->getTranslatableMetadata()->localeProperty;
+        return $this->getPropertyAccessor()->getValue($this->object, $localeProperty);
+    }
+
+    /**
+     * @throws Exception\AnnotationException
+     * @return TranslatableRepositoryInterface
+     */
+    public function getTranslatableRepository()
+    {
+        $meta = $this->getClassMetadata();
+        $repository = $this->objectManager->getRepository($meta->getName());
+
+        if (!($repository instanceof TranslatableRepositoryInterface)) {
+            throw new Exception\AnnotationException(sprintf(
+                'Entity "%s" has "%s" as its "repositoryClass" which does not implement \FSi\DoctrineExtensions\Translatable\Model\TranslatableRepositoryInterface',
+                $meta->getName(),
+                get_class($repository)
+            ));
+        }
+
+        return $repository;
+    }
+
+    /**
+     * @return \Symfony\Component\PropertyAccess\PropertyAccessor
+     */
+    private function getPropertyAccessor()
+    {
+        if (!isset($this->propertyAccessor)) {
+            $this->propertyAccessor = PropertyAccess::createPropertyAccessor();
```

Comment:
I would consider passing one instance of `PropertyAccessor` to every instance of `ObjectAssociationContext`